src/io/parquet/read/binary/basic.rs

```diff
@@ -46,16 +46,14 @@ fn read_dict_buffer<O: Offset>(
     values: &mut Binary<O>,
     validity: &mut MutableBitmap,
 ) {
-    let length = values.len() + additional;
-
     let values_iterator = values_iter(indices_buffer, dict, additional);
 
     let mut validity_iterator = hybrid_rle::Decoder::new(validity_buffer, 1);
 
     extend_from_decoder(
         validity,
         &mut validity_iterator,
-        length,
+        additional,
```

Comment:
parameter mixup; extend_from_decoder expects the additional length, but this was passing total length.